extensions/flight-sql/src/main/java/io/deephaven/server/flightsql/FlightSqlResolver.java

```diff
@@ -0,0 +1,1710 @@
+            final String[] dbSchemaNames = new String[size];
+            final String[] tableNames = new String[size];
+            final String[] tableTypes = new String[size];
+            final Schema[] tableSchemas = includeSchema ? new Schema[size] : null;
+            int count = 0;
+            for (Entry<String, Table> e : queryScopeTables.entrySet()) {
+                final Table table = authorization.transform(e.getValue());
+                if (table == null) {
+                    continue;
+                }
+                final String tableName = e.getKey();
+                if (!tableNameFilter.test(tableName)) {
+                    continue;
+                }
+                catalogNames[count] = null;
+                dbSchemaNames[count] = null;
+                tableNames[count] = tableName;
+                tableTypes[count] = TABLE_TYPE_TABLE;
+                if (includeSchema) {
+                    tableSchemas[count] = BarrageUtil.schemaFromTable(table);
+                }
+                ++count;
+            }
+            final ColumnHolder<String> c1 = TableTools.stringCol(CATALOG_NAME, catalogNames);
+            final ColumnHolder<String> c2 = TableTools.stringCol(DB_SCHEMA_NAME, dbSchemaNames);
+            final ColumnHolder<String> c3 = TableTools.stringCol(TABLE_NAME, tableNames);
+            final ColumnHolder<String> c4 = TableTools.stringCol(TABLE_TYPE, tableTypes);
+            final ColumnHolder<Schema> c5 = includeSchema
+                    ? new ColumnHolder<>(TABLE_SCHEMA, Schema.class, null, false, tableSchemas)
+                    : null;
+            final Table newTable = includeSchema
+                    ? TableTools.newTable(CommandGetTablesConstants.DEFINITION, attributes, c1, c2, c3, c4, c5)
+                    : TableTools.newTable(CommandGetTablesConstants.DEFINITION_NO_SCHEMA, attributes, c1, c2, c3, c4);
+            return count == size
+                    ? newTable
+                    : newTable.head(count);
+        }
+    }
+
+    // ---------------------------------------------------------------------------------------------------------------
+
+    private <Response extends Message> void executeAction(
+            final SessionState session,
+            final ActionHandler<Response> handler,
+            final ActionObserver observer) {
+        // If there was complicated logic going on (actual building of tables), or needed to block, we would instead use
+        // exports or some other mechanism of doing this work off-thread. For now, it's simple enough that we can do it
+        // all on-thread.
+        try {
+            handler.execute(session, new ResultVisitorAdapter<>(observer::onNext));
+        } catch (StatusRuntimeException e) {
+            // We expect other Throwables to be wrapped and transformed if necessary via
+            // io.deephaven.server.session.SessionServiceGrpcImpl.rpcWrapper
+            observer.onError(e);
+            return;
```

Comment:
Discussed in slack, this was a misunderstanding on your part (this in on the RPC-thread, all very well defined unary task _at present_). I updated the comment a bit to make it more clear, and also updated the interface to `StreamObserver<Result>` instead of our own `ActionObserver`. This allows the implementations to use `GrpcUtil.safelyOnNext`, etc.